### DISCUSSION TITLE
How to let output anchors have different values?

### DISCUSSION THREAD

**Discussion by microdou** [open-source-labs/Svelvet#352] [General]
I'd like to let a multi-anchor node behave as follows:
1. Receive parameters from **multiple input anchors**;
2. Compute multiple output values/objects from the inputs;
3. Feed the multiple outputs into **dedicated output anchors**;

I'm currently able to do the first two steps. Is there any suggestion on how to do the third step? Looks like I can only feed the same output store containing all values/objects to every output anchor. How can I let each output anchor only have one dedicated output value/object?

**Comment by briangregoryholmes** [comment]
Would the following work for your use case? (This is obviously simplified)
```HTML
<script>
    const subtract = (inputs: InputStructure) => inputs.value1 - inputs.value2;
    const add = (inputs: InputStructure) => inputs.value1 + inputs.value2;
    const multiply = (inputs: InputStructure) => inputs.value1 * inputs.value2;

    const outputOne = generateOutput(inputs, subtract);
    const outputTwo = generateOutput(inputs, add);
    const outputThree = generateOutput(inputs, multiply);
</script>

<Node >
     <Anchor outputStore={outputOne} output />
     <Anchor outputStore={outputTwo} output />
     <Anchor outputStore={outputThree} output />
</Node>
```

We plan on adding the same idea of a "key" from our input store to output stores as well, which would allow you to make a single processor function that returns an object and just tell each anchor to be associated with a particular key/value pair.

**Reply by microdou** [reply]
Looks like it will work. Thanks!

Was looking for if there is an automated way to extract output values based on key, and looks like that's exactly your plan :)
